src/components/Menu/MenuBar/menuBarData.ts

```diff
@@ -383,52 +383,45 @@ export const useCommonMenuBarData = (store: Store) => {
 export const concatMenuBarData = (
   menuBarContents: MaybeRef<MaybeComputedMenuBarContent>[],
 ): Record<keyof MenuBarContent, MenuItemData[]> => {
-  const indexItems = Object.fromEntries(
-    Object.entries(menuItemStructure).map(([key, value]) => [
-      key as keyof MenuBarCategory,
-      Object.fromEntries(value.map((item) => [item, []])),
-    ]),
-  ) as {
-    [K in MenuBarCategory]: Record<
-      keyof NonNullable<MenuBarContent>[K],
-      MenuItemData[]
-    >;
-  };
+  const result = Object.fromEntries(
+    Object.entries(menuItemStructure).map(([key, value]) => {
+      const singleMenuBar = extractSinglemenuBar(
+        key as MenuBarCategory,
+        menuBarContents,
+      );
+
+      return [
+        key,
+        flatWithSeparator(
+          value
+            .map((menuItemKey) => singleMenuBar[menuItemKey])
+            .filter((v) => v.length > 0),
+          { type: "separator" },
+        ),
+      ];
+    }),
+  ) as Record<keyof MenuBarContent, MenuItemData[]>;
+
+  return result;
+};
 
-  for (const menuBarContent of unref(menuBarContents)) {
-    for (const key in unref(menuBarContent)) {
-      const root = key as MenuBarCategory;
-      const items = unref(unref(menuBarContent)[root]);
-      if (items) {
-        for (const itemKey in items) {
-          // @ts-expect-error 型パズルが大変なので手動で型を合わせる
-          // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-          const item = items[itemKey];
-          if (item) {
-            // @ts-expect-error 型パズルが大変なので手動で型を合わせる
-            // eslint-disable-next-line @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-member-access
-            indexItems[root][itemKey].push(...item);
-          }
-        }
+const extractSinglemenuBar = (
+  key: MenuBarCategory,
+  menuBarContents: MaybeRef<MaybeComputedMenuBarContent>[],
+) => {
+  const itemsArray: Record<string, MenuItemData[]> = {};
```

Comment:
こうかな
```suggestion
  const itemsArray: Record<MenuBarCategory, MenuItemData[]> = {};
```

まあasまみれになるのですが･･･。

あれ、そういえば昔定義してた`objectEntries` `objectFromEntries`ってこの辺使えなかったりしませんかね？